gammacat/webpage.py

```diff
@@ -62,11 +62,25 @@ def make_references_data(self):
         references_data = []
         for reference_id in reference_ids:
             ads_url = f'https://ui.adsabs.harvard.edu/#abs/{reference_id}'
+            year = {reference_id[:4]}
```

Comment:
The bug is here: you're puttiing a Python dict into year, but I think you just want the string, i.e. change to this:
```
year = reference_id[:4]
```

Also, if possible, we should avoid duplicated code that processes locations / urls and fiddles around with things like `reference_id[:4]` that now or in the future might not work for all reference ids.
I don't understand why you need to extract the year here, when we managed to avoid it for the source detail pages.
See `make_resource_info_for_template` as example.
The location should only be generated once when making the output index file, and after that you should use that existing field to make derived things like locations in the docs folder or URLs.